cpp/src/io/parquet/writer_impl.cu

```diff
@@ -85,6 +87,28 @@ parquet::Compression to_parquet_compression(compression_type compression)
   }
 }
 
+size_type column_size(column_view const& column, rmm::cuda_stream_view stream)
+{
+  if (column.num_children() == 0) { return size_of(column.type()) * column.size(); }
+
+  if (column.type().id() == type_id::STRING) {
+    auto scol         = strings_column_view(column);
+    size_type colsize = cudf::detail::get_value<size_type>(scol.offsets(), column.size(), stream);
```

Comment:
what if there are sliced rows at the start of a column. I think we would need to subtract the first offset from this.

Reply:
I'm frankly amazed this code worked at all :wink: Yes, I'll subtract `offsets[0]`.